challenge_2/haskell/halogenandtoast/src/Challenge.hs

```diff
@@ -0,0 +1,21 @@
+module Challenge where
+
+import Data.Maybe ( fromMaybe )
+import Data.HashMap.Strict ( alter
+                           , toList
+                           , filterWithKey
+                           , empty
+                           , HashMap(..)
+                           )
+
+counts :: [String] -> HashMap String Int
```

Comment:
I would alter this to be generically applicable.
counts :: (Eq a, Hashable a) => [a] -> HashMap a Int